NDTensors/ext/NDTensorsCUDAExt/mul.jl

```diff
@@ -0,0 +1,11 @@
+function LinearAlgebra.mul!(
+  C::Exposed{<:CuArray,<:LinearAlgebra.Transpose},
+  A::Exposed{<:CuArray},
+  B::Exposed{<:CuArray},
+  α,
+  β,
+)
+  return mul!(
+    expose(parent(C)), expose(transpose(unexpose(B))), expose(transpose(unexpose(A))), α, β
+  )
```

Comment:
```suggestion
  mul!(
    expose(parent(C)), expose(transpose(unexpose(B))), expose(transpose(unexpose(A))), α, β
  )
  return unexpose(C)
```
Like in other PRs, for in-place functions we should use the convention that we return the input value that is being modified (in this case the unexposed value since we always unexpose before returning). The way you wrote it, it would output the non-transposed version of `C`. Please try to be mindful about this in the future, I've corrected this a number of times in previous PRs.

Reply:
Okay I will make sure to remember this going forward, thanks!

Reply:
Question regarding this, so in my test I do something like `C(3,1)^T = A(1,2) * B(2,3)`. My understanding here was that C is only being transposed in the contraction so that it can be efficiently added to the result of `A(1,2) * B(2,3)`, i.e. `C(1,3) = 1 .* (A * B) (1,3) + 0 .* C(1,3)` but we are actually interested in `C(3,1)`. So for example in my test below I expect C to have the dimensions `(4x2)` not `(2x4)`, is this correct? 
In the same direction of logic, here `parent(C)` is in the desired index order so I transpose the `(A * B)` portion of the equation such that 
```
(C(3,1))^T = (A * B)(1,3) ==> C(3,1) = (B^T * A^T)(3,1)
```
I just want to be sure I have this right, thanks for the help Matt!

Reply:
I'm not really following, I think writing a test which I can review would be easier.